src/VecSim/algorithms/hnsw/hnsw.h

```diff
@@ -1468,15 +1467,17 @@ VecSimQueryResult *HNSWIndex<DataType, DistType>::searchRangeBottomLayer_WithTim
         // requested radius.
         // Here we send the radius as double to match the function arguments type.
         processCandidate_RangeSearch(curr_el_pair.second, data_point, 0, epsilon, visited_tag,
-                                     &results, candidate_set, dynamic_range_search_boundaries,
+                                     res_container, candidate_set, dynamic_range_search_boundaries,
                                      radius);
     }
 #ifdef ENABLE_PARALLELIZATION
     visited_nodes_handler_pool->returnVisitedNodesHandlerToPool(this->visited_nodes_handler);
 #endif
 
     *rc = VecSim_QueryResult_OK;
-    return results;
+    auto res = res_container->get_results();
```

Comment:
no need to worry about memory leaks if res_container is a shared_ptr
and then
return res_container->get_results();

Reply:
But isn't it heavier on performance?

Reply:
im not sure that the heavy part is fetching the data (which we do multiple times) and not the ctor (which happens once)
but its much safer 
maybe we can bm it?

Reply:
I agree it is safer, but for temporary usage in a single function I think it is a bit redundant